### ISSUE TITLE
Is there way to change nodes margin and padding not inline?

### ISSUE THREAD

**Issue by lokhmakov** [tokarchyn/react-easy-diagram#10]
`.react_fast_diagram_DiagramInner *` override any changes.

**Comment by tokarchyn** [comment]
Right now customization using css is in bad shape and this is not possible😔 But, I'm thinking about simple solution by providing "nocss" version of library, where it will be possible to copy css from the library, modify it, and just import like all other css files. Will it solve your problems?

In the future I am intend to provide one more option for this, that will not require working with css file, but I don't think I will have time for this for a month.

**Comment by lokhmakov** [comment]
Also problem to override Diagram component due to hardcoded RootStoreContext.

May be some examples how to define custom ports before render <Port /> component.

In some cases no need absolute for Ports.

Sorry for hints mess! :)

**Comment by lokhmakov** [comment]
Yes u can remove import of CSS in library and provide it in dist and add to Getting started, something like:

```js
import 'react-easy-diagram/react-easy-diagram.min.css'
```

**Comment by tokarchyn** [comment]
It's ok:)

What do you mean by overriding Diagram? What is the purpose of this?

I will write some docs for Ports customization on weekend. 
In short, if you would like to place Ports by yourself wherever you want, all you need to do is to use Port component without position specified  in its props, in this case library will not add to Port style properties like "position: absolute", transform, left etc. 

To use your own Port component (Node can be also replaced by your own component in similar way) just add it to settings like this:
```jsx
const MyCustomPort = (props) => {
  return <div>x</div>;
}

  <Diagram
    settings={{
      ports: {
        portComponents: {
          default: MyCustomPort // this way all ports by default will be rendered with your component
        },
      },
    }}
  />
```
But if you want also to be able to read from port state and rerender if something changed, you should wrap your component into `observer` (it's from mobx library, the state management used in react-easy-diagram):
```jsx
import { observer } from 'mobx-react-lite';

const MyCustomPort = observer((props: any) => {
  return props.entity.hovered ? <div>X</div> : <div>x</div>; // now whenever user hover over the port, port component will be rerendered and display big "X" instead of "x"
})
```

**Comment by lokhmakov** [comment]
> What do you mean by overriding Diagram? What is the purpose of this?

It's related to case where u can rewrite Diagram and avoid Diagram.css import! But u cannot due to hardcoded Context and import it from (./hooks/useRootStore.ts). Will be solved with nocss version! :)

**Comment by tokarchyn** [comment]
Thank you for explaining!

**Comment by tokarchyn** [comment]
There will be a new version of library as soon as I write the documentation and examples, since there are several breaking changes in styling built-in components. Unfortunately, I couldn't do it this weekend.

**Comment by tokarchyn** [comment]
New version has been published, feel free to check it out:)
It enables to:
- add your own classes and styles that will override default ones
- disable default classes
- '.react_fast_diagram_DiagramInner *' class was removed
- your own classes take precedence over default one, as css classes are injected at top of <head>

There are also docs on how to customize nodes, ports an links components or provide your own.
